common/src/service/ExceptionHandlingHelper.cs

```diff
@@ -5,29 +5,53 @@ namespace Microsoft.Azure.Devices
 {
     using System;
     using System.Collections.Generic;
+    using System.Globalization;
     using System.Net;
     using System.Net.Http;
     using System.Threading.Tasks;
-
+    using Microsoft.Azure.Devices.Common;
     using Microsoft.Azure.Devices.Common.Exceptions;
+    using Microsoft.Azure.Devices.Common.Extensions;
+    using Microsoft.Azure.Devices.Shared;
+    using Newtonsoft.Json;
 
     internal class ExceptionHandlingHelper
     {
         public static IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> GetDefaultErrorMapping()
         {
             var mappings = new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>();
 
-            mappings.Add(HttpStatusCode.NoContent, async (response) => new DeviceNotFoundException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false), innerException: null));
-            mappings.Add(HttpStatusCode.NotFound, async (response) => new DeviceNotFoundException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false), innerException: null));
-            mappings.Add(HttpStatusCode.Conflict, async (response) => new DeviceAlreadyExistsException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false), innerException: null));
-            mappings.Add(HttpStatusCode.BadRequest, async (response) => new ArgumentException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false)));
-            mappings.Add(HttpStatusCode.Unauthorized, async (response) => new UnauthorizedException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false)));
-            mappings.Add(HttpStatusCode.Forbidden, async (response) => new QuotaExceededException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false)));
-            mappings.Add(HttpStatusCode.PreconditionFailed, async (response) => new DeviceMessageLockLostException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false)));
-            mappings.Add(HttpStatusCode.RequestEntityTooLarge, async (response) => new MessageTooLargeException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false))); ;
-            mappings.Add(HttpStatusCode.InternalServerError, async (response) => new ServerErrorException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false)));
-            mappings.Add(HttpStatusCode.ServiceUnavailable, async (response) => new ServerBusyException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false)));
-            mappings.Add((HttpStatusCode)429, async (response) => new ThrottlingException(message: await GetExceptionMessageAsync(response).ConfigureAwait(false)));
+            mappings.Add(HttpStatusCode.NoContent, async (response) => new DeviceNotFoundException(await GetExceptionCodeAsync(response).ConfigureAwait(false),
```

Comment:
I would use named args here as well.

Reply:
yupp I added those just one commit ago to fix a bug. It would be safer to keep the named args.
Also this is reverting the fix for #599 - There is no overload in DeviceNotFoundException for a code and message. It expects a deviceId and IotHubName and constructs a message, which will end up looking weird.

Reply:
Nevermind you added an overload below.